From 85e224c603912b8c0571c3b19760ed43340a3e84 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 25 Oct 2018 18:07:54 +0800
Subject: [PATCH 4924/5242] MLK-20056-2 ARM64: dts: imx8qm: passthrough hifi
 dsp to domu

commit  c5444808358e34fd3e501257768b6417e549da37 from
https://source.codeaurora.org/external/imx/linux-imx.git

Passthrough hifi dsp to DomU, smmu is enabled for dsp,
so the mu13_B side also needs to be mapped into smmu table.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |   19 ++++++++++++++++++-
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |   20 +++++++++++++++++++-
 arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi  |    7 +++++++
 3 files changed, 44 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index f1978b8..af61cbe 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -167,6 +167,12 @@
 
 				SC_R_MCLK_OUT_0
 				SC_R_MCLK_OUT_1
+
+				/* HIFI DSP */
+				SC_R_DSP
+				SC_R_DSP_RAM
+				SC_R_MU_13B
+				SC_R_MU_13A
 			>;
 			pads = <
 				/* i2c1_lvds1 */
@@ -284,6 +290,11 @@
 		xen,passthrough;
 	};
 
+	dsp_mu_b: dsp_mu_b@0x5d310000 {
+		reg = <0x0 0x5d310000 0x0 0x10000>;
+		xen,passthrough;
+	};
+
 	decoder_boot_mem: decoder_boot_mem@0x84000000 {
 		xen,passthrough;
 		reg = <0 0x84000000 0 0x2000000>;
@@ -419,7 +430,7 @@
 		      <&usdhc1 0x12>, <&usbotg1 0x11>,
 		      <&edma01 0x10>, <&cm41 0x09>, <&pciea 0x08>,
 		      <&vpu_decoder 0x7>, <&crypto 0x6>, <&isi_0 0x5>,
-		      <&usbotg3 0x4>;
+		      <&usbotg3 0x4>, <&dsp 0x3>;
 };
 
 &lvds_region2 {
@@ -614,6 +625,12 @@
 
 &dsp {
        xen,passthrough;
+       #stream-id-cells = <1>;
+       iommus = <&smmu>;
+};
+
+&mu13 {
+	xen,passthrough;
 };
 
 &mu_m0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index 8284806..c6b3939 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -119,6 +119,11 @@
 			reg = <0 0x91400000 0 0x1000000>;
 		};
 
+		dsp_reserved: dsp@0x92400000 {
+			no-map;
+			reg = <0 0x92400000 0 0x2000000>;
+		};
+
 		decoder_str: str@0x94400000 {
 			no-map;
 			reg = <0 0x94400000 0 0x1800000>;
@@ -236,6 +241,15 @@
 			status = "okay";
 		};
 
+		mu13: mu13@5d280000 {
+			compatible = "fsl,imx8-mu-dsp";
+			reg = <0x0 0x5d280000 0x0 0x10000>;
+			interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
+			fsl,dsp_ap_mu_id = <13>;
+			status = "okay";
+		};
+
+
 		vpu_decoder: vpu_decoder@2c000000 {
 			compatible = "nxp,imx8qm-b0-vpudec", "nxp,imx8qxp-b0-vpudec";
 			boot-region = <&decoder_boot>;
@@ -548,7 +562,11 @@
 /delete-node/ &dpr2_channel3;
 /delete-node/ &dpu1;
 /delete-node/ &pixel_combiner1;
-/delete-node/ &dsp;
+
+&dsp {
+	status = "okay";
+};
+
 /delete-node/ &irqsteer_dsi1;
 /delete-node/ &i2c0_mipi_dsi1;
 /delete-node/ &mipi_dsi_csr2;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
index 632d25d..7d9e4ea 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
@@ -374,6 +374,13 @@
 	aud_rec_clk1_lpcg: aud_rec_clk1_lpcg {
 		reg = <0x0 0x59d10000 0x0 0x1000>;
 	};
+
+	aud_dsp_lpcg: aud_dsp_lpcg {
+		reg = <0x0 0x59580000 0x0 0x1000>;
+	};
+	aud_ocram_lpcg: aud_ocram_lpcg {
+		reg = <0x0 0x59590000 0x0 0x1000>;
+	};
 };
 
 /delete-node/ &edma0;
-- 
1.7.9.5

